types/servicenow/index.d.ts

```diff
@@ -0,0 +1,587 @@
+    add(value: GlideDuration): GlideDuration;
+    getByFormat(format: string): string;
+    getDayPart(): number;
+    getDisplayValue(): string;
+    getDurationValue(): string;
+    getNumericValue(): number;
+    getRoundedDayPart(): number;
+    getValue(): string;
+    setDisplayValue(asDisplayed: string): void;
+    setValue(o: object): void;
+    subtract(value: GlideDuration): GlideDuration;
+}
+
+declare const GlideFilter: {
+    checkRecord(gr: servicenow.GlideRecord, filter: string, matchAll?: object): boolean;
+};
+
+declare const GlideGuid: {
+    generate(): string;
+};
+
+declare class GlideLocale {
+    static get(): GlideLocale;
+    getGroupingSeparator(): string;
+    getDecimalSeparator(): string;
+}
+
+declare const GlidePluginManager: {
+    isActive(pluginId: string): boolean;
+};
+
+declare class GlideSchedule {
+    constructor();
+    constructor(sysId: string, timeZone: string);
+    add(startDate: GlideDateTime, offset: GlideDuration): GlideDateTime;
+    duration(startDate: GlideDateTime, endDate: GlideDateTime): GlideDuration;
+    getName(): string;
+    isInSchedule(time: GlideDateTime): string;
+    isValid(): boolean;
+    load(sysId: string, timeZone: string, excludeSpanId: string): void;
+    setTimeZone(tz: string): void;
+    whenNext(time: GlideDateTime, timeZone: string): number;
+}
+
+declare class GlideEvaluator {
+    constructor();
+    evaluateScript(grObj: servicenow.GlideRecord, scriptField?: string, variables?: object): object;
+    getVariable(name: string): {};
+    putVariable(name: string, value: {}): void;
+}
+
+declare const GlideSecureRandomUtil: {
+    getSecureRandomInt(): number,
+    getSecureRandomIntBound(bound: number): number,
+    getSecureRandomLong(): number,
+    getSecureRandomString(length: number): string
+};
+
+declare class GlideSysAttachment {
+    constructor();
+    copy(sourceTable: string, sourceSysId: string, destinationTable: string, destinationSysId: string): void;
+    deleteAttachment(sysId: string): void;
+    getContent(record: servicenow.GlideRecord): any;
+    getContentBase64(record: servicenow.GlideRecord): string;
+    getContentStream(sysId: string): object;
+    write(record: servicenow.GlideRecord, fileName: string, contentType: string, data: any): string;
+    writeBase64(record: servicenow.GlideRecord, fileName: string, contentType: string, base64Content: string): string;
+    writeContentStream(record: servicenow.GlideRecord, fileName: string, contentType: string, inputStream: object): string;
+}
+
+declare namespace servicenow {
+    interface GlideRecord {
+        readonly sys_created_by: string & GlideElement;
+        readonly sys_created_on: GlideDateTime & GlideElement;
+        readonly sys_id: string;
+        readonly sys_mod_count: number & GlideElement;
+        readonly sys_updated_by: string & GlideElement;
+        readonly sys_updated_on: GlideDateTime & GlideElement;
+        new (type: string): GlideRecord;
```

Comment:
So `GlideRecord` describes the static side of a class, which when constructed returns... a `GlideRecord`? Are you sure?